### PR TITLE
S28 2877: MK Playback Endpoint with JWT

### PR DESCRIPTION
**Pull request by lucas-phillips28**
### JIRA ticket(s)
- S28-2877

### Change description
- Add: GET `/media-service/vod` endpoint

> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Look on mediakind, see there is NOT a content key policy with a name matching your user id. Call endpoint, see one is created. 
- Call endpoint for valid asset, on asset you are trying to watch, see a new locator with a name that matches the user's id. See start time is now, end time is one hours time. Call the endpoint again after one minute. See times update. 
- Check response works for viewing the asset (see steps below) 
- Call endpoint for a recording that the user does not have access to
- Call endpoint for a recording that does not exist
- Call endpoint for a recording that has been marked as deleted
- On MediaKind, turn off `default` endpoint. See when endpoint called, the streaming endpoint is turned back on.  

**How to see if response object values work for playback using MKPlayer (`HLS`):**
1. Select `Hls` option on MKPlayer
2. Select `ClearKey`
3. Copy `hls_url` value from endpoint response and paste in Source URL. 
4. Copy `token` value from endpoint response and set value of License request headers to `{ "Authorization": "Bearer <token>" }`
5. Click Play (Should start to play)

**How to see if response object values work for playback using MKPlayer (`DASH`):**
1. Select `Dash` option on MKPlayer
2. Select `ClearKey`
3. Copy `dash_url` value from endpoint response and paste in Source URL. 
4. Copy `license_url` value from endpoint response and paste in License URL.
5. Copy `token` value from endpoint response and set value of License request headers to `{ "Authorization": "Bearer <token>" }`
6. Click Play (Should start to play)

### CODE REVIEW

File: charts/pre-api/values.yaml [hunk 1]
```diff
@@ -42,3 +44,4 @@ java:
     AZURE_INGEST_SA: "preingestsa{{ .Values.global.environment }}"
     AZURE_FINAL_SA: "prefinalsa{{ .Values.global.environment }}"
     PLATFORM_ENV_TAG: Production
+    ISSUER: https://sts.windows.net/531ff96d-0ae9-462a-8d2d-bec7c0b42082/
```
Comment (jasonpaige) [inline charts/pre-api/values.yaml:47]: I assume this is a tenant level thing so consistent across all envs?

File: src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java [hunk 2]
```diff
@@ -89,6 +96,24 @@ public ResponseEntity<LiveEventDTO> getLiveEvents(@PathVariable String liveEvent
...
+    public ResponseEntity<PlaybackDTO> getVod(
+        @RequestParam UUID recordingId,
+        @RequestParam(required = false) String mediaService
+    ) throws InterruptedException {
+        // check recording exists + authed
+        recordingService.findById(recordingId);
+
+        var service = mediaServiceBroker.getEnabledMediaService(mediaService);
+
+        // TODO dont rely on naming convention, link asset name in db
```
Comment (jasonpaige) [inline src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java:110]: Not for this PR but maybe we need to consider some kind of naming utility class to centralise these naming conventions (same for shortened/sanitised UUIDs)

File: src/main/java/uk/gov/hmcts/reform/preapi/media/MediaKindClient.java [hunk 3]
```diff
@@ -33,11 +36,14 @@ public interface MediaKindClient {
     @PutMapping("/liveEvents/{liveEventName}")
     MkLiveEvent putLiveEvent(@PathVariable String liveEventName, @RequestBody MkLiveEvent mkLiveEvent);
 
-    @GetMapping("/liveEvents")
-    MkGetListResponse<MkLiveEvent> getLiveEvents(@RequestParam("$skipToken") int skipToken);
+    @PostMapping("/assets/{assetName}/listStreamingLocators")
```
Comment (jasonpaige) [inline src/main/java/uk/gov/hmcts/reform/preapi/media/MediaKindClient.java:39]: This is a POST?? 🤦

**Review by jasonpaige — APPROVED** [review]

File: charts/pre-api/values.yaml (hunk 1 above)
Comment (lucas-phillips28) [reply charts/pre-api/values.yaml:47]: Yeah I believe so- at least seems to be the same value in stg, test and demo envs.

File: src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java (hunk 2 above)
Comment (lucas-phillips28) [reply src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java:110]: Yeah definitely agree with this

**Review by NazeeKadiu — DISMISSED** [review]
Testing completed and passed.

**Review by NazeeKadiu — DISMISSED** [review]
Looks good

**Pull request merged**